uk_geo_utils/helpers.py

```diff
@@ -169,6 +169,108 @@ def __str__(self):
         return ",".join(self.generate_address_label())
 
 
+class LocalAuthAddressFormatter:
+    def __init__(
+        self,
+        organisation_name,
+        sao_start_number,
+        sao_start_suffix,
+        sao_end_number,
+        sao_end_suffix,
+        sao_text,
+        pao_start_number,
+        pao_start_suffix,
+        pao_end_number,
+        pao_end_suffix,
+        pao_text,
+        street_description,
+        locality,
+        town_name,
+    ):
+        self.organisation_name = organisation_name
+        self.sao_start_number = sao_start_number
+        self.sao_start_suffix = sao_start_suffix
+        self.sao_end_number = sao_end_number
+        self.sao_end_suffix = sao_end_suffix
+        self.sao_text = sao_text
+        self.pao_start_number = pao_start_number
+        self.pao_start_suffix = pao_start_suffix
+        self.pao_end_number = pao_end_number
+        self.pao_end_suffix = pao_end_suffix
+        self.pao_text = pao_text
+        self.street_description = street_description
+        self.locality = locality
+        self.town_name = town_name
+
+    @property
+    def primary_addressable_object(self):
+        return self._get_addressable_object(
+            self.pao_text,
+            self.pao_start_number,
+            self.pao_start_suffix,
+            self.pao_end_number,
+            self.pao_end_suffix,
+        )
+
+    @property
+    def secondary_addressable_object(self):
+        return self._get_addressable_object(
+            self.sao_text,
+            self.sao_start_number,
+            self.sao_start_suffix,
+            self.sao_end_number,
+            self.sao_end_suffix,
+        )
+
+    def _get_addressable_object(
+        self, text, start_number, start_suffix, end_number, end_suffix
+    ):
+        ao = ""
+        if text:
+            ao = ao + text + " "
+
+        if start_number and not start_suffix and not end_number:
+            ao = ao + start_number + " "
+        else:
+            ao = ao + start_number
+
+        if start_suffix and not end_number:
+            ao = ao + start_suffix + " "
+        elif start_suffix and end_number:
+            ao = ao + start_suffix
+
+        if end_suffix and end_number:
+            ao = ao + "-"
+        elif start_number and end_number:
+            ao = ao + "-"
+
+        if end_number and not end_suffix:
+            ao = ao + end_number + " "
+        elif not end_number:
+            ao = ao + ""
+        else:
+            ao = ao + end_number
+
+        if end_suffix:
+            ao = ao + end_suffix + " "
+
+        return ao
```

Comment:
This logic looks like it's based off of page 53 of [this](https://www.ordnancesurvey.co.uk/documents/addressbase-products-getting-started-guide1.pdf). Is it worth sticking in a link so anyone looking at this in the future knows to go and look for such a document?